src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/AuditTrailTrimmingBackgroundTask.cs

```diff
@@ -0,0 +1,67 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.AuditTrail.Settings;
+using OrchardCore.BackgroundTasks;
+using OrchardCore.Entities;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OrchardCore.AuditTrail.Services
+{
+    [BackgroundTask(Schedule = "*/10 * * * *", Description = "A background task that regularly deletes old Audit Trail Events.")]
```

Comment:
Because the unit of retention is days (which I think is good) the schedule of the task should be about the same. I don't think we need to run it more than once a day. That will prevent from running too many SQL queries.

Reply:
Or run it every hour, because the unit of the minimum run interval is in hours?

Reply:
`0 0 */1 * *` => `At 00:00 on every day-of-month.`

Check it on https://crontab.guru